pkg/providers/gcp/provider_postgres.go

```diff
@@ -709,3 +764,29 @@ func buildDefaultCloudSQLSecret(p *v1alpha1.Postgres) (*v1.Secret, error) {
 		Type: v1.SecretTypeOpaque,
 	}, nil
 }
+
+func checkUpdateDenyPeriod(ctx context.Context, sqladminService gcpiface.SQLAdminService,
+	projectID string, instance *sqladmin.DatabaseInstance) error {
+	// if current Time(date) minus lastMaintenanceTime < 90 days (2160 hours without few hours = 2150), do nothing,
+	// otherwise - update Maintenance Deny period
+	currentTime := time.Now()
+	if currentTime.Sub(lastMaintenanceTime) < 2150 {
```

Comment:
Do we have to save `lastMaintenanceTime`? can we not check the `EndDate` on the instance and compare to current time?

Reply:
compare EndDate on the instance with  current time?
It looks very good idea!
Thank you very much